src/modules/job-info/job_state.c

```diff
@@ -1745,45 +1677,126 @@ static int job_events_annotations (struct job_state_ctx *jsctx,
     return 0;
 }
 
-static void job_events_continuation (flux_future_t *f, void *arg)
+static int job_events_process (struct job_state_ctx *jsctx, json_t *events)
 {
-    struct job_state_ctx *jsctx = arg;
     size_t index;
     json_t *value;
-    json_t *events;
-
-    if (flux_rpc_get_unpack (f, "{s:o}", "events", &events) < 0) {
-        flux_log_error (jsctx->h, "%s: flux_rpc_get_unpack", __FUNCTION__);
-        goto error;
-    }
-
-    if (!json_is_array (events)) {
-        flux_log (jsctx->h, LOG_ERR, "%s: events EPROTO", __FUNCTION__);
-        goto error;
-    }
 
     json_array_foreach (events, index, value) {
         flux_jobid_t id;
+        int eventlog_seq;
         json_t *entry;
-        const char *name;
         double timestamp;
+        const char *name;
         json_t *context = NULL;
 
-        if (json_unpack (value, "{s:I s:o}", "id", &id, "entry", &entry) < 0
+        if (json_unpack (value, "{s:I s:i s:o}",
+                                "id", &id,
+                                "eventlog_seq", &eventlog_seq,
+                                "entry", &entry) < 0
             || eventlog_entry_parse (entry, &timestamp, &name, &context) < 0) {
             flux_log (jsctx->h, LOG_ERR, "%s: error parsing record",
                       __FUNCTION__);
-            goto error;
+            return -1;
         }
 
-        if (!strcmp (name, "priority")) {
-            if (job_events_priority (jsctx, id, timestamp, context) < 0)
-                goto error;
+        if (!strcmp (name, "submit")) {
+            if (job_events_submit (jsctx,
```

Comment:
The long set of if/elses should probably be broken out of the for loop into a separate function for readability.

Any refactoring possible here?  There's a lot of repetition, and some function calls called here are just wrappers for other functions.

Reply:
> Any refactoring possible here?

I couldn't come up with something simple/obvious here, as many of the function calls are just different enough.  But now that you mention it, I can remove some of the dumb "wrapping" function calls, which might make things look simpler.